src/core_plugins/kibana/public/dashboard/panel/panel_utils.js

```diff
@@ -63,4 +66,18 @@ export class PanelUtils {
     }, 0);
     return ++maxId;
   }
+
+  static getPanelIndexPatterns(panels, objectLoadersForDashboard) {
+    return Promise.map(panels, (panel) => {
+      return loadSavedObject(objectLoadersForDashboard, panel)
+        .then(({ savedObj }) => {
+          if (savedObj.vis) {
+            return savedObj.vis.indexPattern;
+          } else if (savedObj.searchSource) {
+            return savedObj.searchSource.get('index');
+          }
+        });
+    })
+      .then(indexPatterns => _.uniq(indexPatterns));
+  }
```

Comment:
@Stacey-Gammon Could you take a look at this portion? Is there a better way to get a unique list of index patterns used by the panels in a dashboard?